### ISSUE TITLE
dev: improve testing for block

### ISSUE THREAD

**Issue by greged93** [keep-starknet-strange/madara#639]
The Starknet pallet function currently uses two functions `store_block` and `get_block_context` which are not unit tested. Add unit testing for these functions and organise these into a block related test file. You should also move to there any block related tests we have in the Starknet pallet (e.g. `given_normal_conditions_when_current_block_then_returns_correct_block`).

**Comment by chirag-bgh** [comment]
I can take this.

**Comment by greged93** [comment]
hey @chirag-bgh we try to keep one issue at the time :)

**Comment by greged93** [comment]
Assigned because chirag is blocked on other issues

**Comment by chirag-bgh** [comment]
Blocked by #329

**Comment by abdelhamidbakhta** [comment]
> Blocked by #329

Can you expand on why it's blocking this one ?

**Comment by chirag-bgh** [comment]
> > Blocked by #329
> 
> Can you expand on why it's blocking this one ?

Not the whole part but only for `get_block_context()`. In `get_block_context()` function we need it set the gas_price based on the L1 gas price.  This can be done after #611 is merged

**Comment by chirag-bgh** [comment]
> > > > Blocked by #329
> > > 
> > > 
> > > Can you expand on why it's blocking this one ?
> > 
> > 
> > Not the whole part but only for `get_block_context()`. In `get_block_context()` function we need it set the gas_price based on the L1 gas price. This can be done after #611 is merged
> 
> Can you still move forward on what is not blocked ? And later we can do another issue for the remaining part

Sure ser!

**Comment by nikania** [comment]
If no one is working on this, I can take it

**Comment by abdelhamidbakhta** [comment]
> If no one is working on this, I can take it

please do, assigned to you
